File: io/lucamienert/basic/ast/LetStatement.java
package io.lucamienert.basic.ast;

import io.lucamienert.basic.instruction.Instruction;
import io.lucamienert.basic.instruction.InstructionSequence;
import io.lucamienert.basic.instruction.OpCode;

public class LetStatement extends Statement {

    private final String name;
    private final Expression value;

    public LetStatement(String name, Expression value) {
        this.name = name;
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public Expression getValue() {
        return value;
    }

    @Override
    public void compile(InstructionSequence seq) {
        value.compile(seq);
        seq.append(new Instruction(OpCode.STORE, name));
    }
    
}

File: BasicCompiler/Ast/LetStatement.cs
using BasicCompiler.Instructions;

namespace BasicCompiler.Ast;

public class LetStatement : Statement
{
    public string Name { get; set; }
    public Expression Value { get; set; }

    public LetStatement(string name, Expression value)
    {
        Name = name;
        Value = value;
    }

    public override void Compile(InstructionSequence instructionSequence)
    {
        Value.Compile(instructionSequence);
        instructionSequence.Append(new Instruction(OpCode.STORE, Name));
    }
}
